templates/nextflow-launch.yaml

```diff
@@ -0,0 +1,204 @@
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: batch.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole
+  InstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2008-10-17'
+        Statement:
+          - Sid: ''
+            Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role
+        - arn:aws:iam::aws:policy/AmazonS3FullAccess
+        - !Ref JobManagedPolicy
+  SpotFleetRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2008-10-17'
+        Statement:
+          - Sid: ''
+            Effect: Allow
+            Principal:
+              Service: spotfleet.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonEC2SpotFleetTaggingRole
+  ComputeEnvironmentOnDemand:
+    Type: AWS::Batch::ComputeEnvironment
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3008
+    Properties:
+      Type: MANAGED
+      State: ENABLED
+      ServiceRole: !Ref BatchServiceRole
+      ComputeResources:
+        Type: EC2
+        MaxvCpus: !Ref MaxvCpus
+        Subnets: !Ref SubnetIds
+        SecurityGroupIds:
+          - !Ref EcsSecurityGroupId
+        InstanceRole: !Ref InstanceRole
+        LaunchTemplate:
+          LaunchTemplateId: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplate'
+          LaunchTemplateName: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplate'
+          Version: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplateLatestVersionNumber'
+  ComputeEnvironmentSpot:
+    Type: AWS::Batch::ComputeEnvironment
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3008
+    Properties:
+      Type: MANAGED
+      State: ENABLED
+      ServiceRole: !Ref BatchServiceRole
+      ComputeResources:
+        Type: SPOT
+        MaxvCpus: !Ref MaxvCpus
+        Subnets: !Ref SubnetIds
+        SecurityGroupIds:
+          - !Ref EcsSecurityGroupId
+        SpotIamFleetRole: !Ref SpotFleetRole
+        InstanceRole: !Ref InstanceRole
+        LaunchTemplate:
+          LaunchTemplateId: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplate'
+          LaunchTemplateName: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplate'
+          Version: !ImportValue
+            'Fn::Sub': '${AWS::Region}-nextflow-ecs-cluster-EcsLaunchTemplateLatestVersionNumber'
+  JobQueueOnDemand:
```

Comment:
these on demand and spot queues are separate and requires users to specify which queue to put their jobs on.  I assume NF tower has a way for users to specify this either using the NF CLI or console?

Reply:
Correct, there is a way to do this within the console and it looks like this:

<img width="1078" alt="image" src="https://user-images.githubusercontent.com/10556275/231035213-ed006202-0f2a-4648-96bf-1b6c8559cb7a.png">